tiling.js

```diff
@@ -205,6 +207,59 @@ export function disable () {
     backgroundSettings = null;
     interfaceSettings = null;
     workspaceSettings = null;
+
+    unpatchMonitorInFullscreen();
+}
+
+// Patch monitor objects prototype to check our space for the inFullscreen
+// property.
+function patchMonitorInFullscreen() {
```

Comment:
@Lythenas, that this type of patching needs to be done in patches.js and using the patterns there, otherwise this wouldn't pass EGO review) - see examples there for patching prototypes (which will then be automatically reverted on PaperWM disable).

But in principal I can see what you've done here.

Reply:
All good though - we can do that after.

Reply:
Didn't know about the `patches.js` file. Will move it there.

But I think we can't use the pattern there, since `monitor.prototype` is undefined. Also I am not overwriting a normal function or property. I am setting a getter function for a property. And since that does not exist by default there is nothing to safe.

The `monitor` objects are also just plain JS objects as far as I can tell, so no class instance. The prototype seems to apply to all monitor objects but no other objects for some reason. I also don't know what happens when you plug in or unplag  a monitor afterwards or change the monitor configuration. It is possible that the prototype will be lost when the objects change.

Maybe there is a better way to patch this but that is what I found works.

Reply:
Also note that fullscreen avoider does some interesting stuff for patching by calling `toString` on the function and string replacing some things: https://github.com/Noobsai/fullscreen-avoider/blob/master/src/extension.js#L91

Reply:
I see, given what we're doing in this case, we can add it into `Utils.js` `upgradeGnomeMonitors`:

https://github.com/paperwm/PaperWM/blob/release/utils.js#L526-L536

This is nice since it's a dbus service that will get called whenever monitors change (e.g. if we add new monitors and remove monitors).  We use this to upgrade monitor objects to add connector information (e.g. HDMI-1, eDP-1 etc.).

Reply:
> Edit: I think the code in `patches` works directly on classes, right? So I don't think we can use it since `monitor` are just plain JS objects as far as I can tell.

No, `Monitor` is indeed a class - it's here:
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/layout.js?ref_type=heads#L158-L171

You're essentially overriding the prototype class getter `get inFullscreen()`.

Interestingly, Gnome haven't exported this so that's the main reason why we can't use it directly in patches.  You're approach of getting a concrete implementation of one and overriding the prototype (which effects the Monitor class) will work though.